src/js/Events.js

```diff
@@ -35,13 +35,21 @@ function getmover(mouse) {
             dy = yy - mouse.y;
             var ref = Math.sqrt(dx * dx + dy * dy);
             dist = ref - rad.value.real;
-            dx = 0;
-            dy = 0;
+
+            if (ref > 0) {
```

Comment:
I don't get what's going on here?

Reply:
It checks whether the distance of the mouse coordinates to the reference point on the circle is larger than 0, I guess. But it looks a bit strange to me, too.

Reply:
`dx` and `dy`  should store the vector of the mouse to the actual thing that is moved (compare, for instance, the branch for `if (el.kind === "P")`).

In the `if (el.kind === "C")`-branch, `dx` and `dy` is first "wrongly" initialized as the vector of the mouse to the center of the circle (We use these values to compute `dist`). However, we do not move the center of circle, the reference point is supposed to a point on the circle (which has the coordinates `(xx-dx/ref*rad.value.real,xx-dx/ref*rad.value.real)`, the projection of the mouse to the circle ). In the `if (ref > 0) {`-branch I update `dx`/`dy` accordingly. `ref>0` is just checked to avoid division by zero.
If you find this code bad to read (I am also not really happy, but I tried to be minimally invasive ), I can rewrite this branch.